File: Dart/1-50/17.dart
import 'dart:io';

int fff(int n) {
  int s = 0, p;
  while (n > 0) {
    p = n % 10;
    s = s + p;
    n = n ~/ 10;
  }
  return s * s;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  int k = 0;
  int i = 1, h = 0;
  while (k < n) {
    if (i % fff(i) == 0)
      k++;
    h = i;
    i++;
  }
  print(h);
}

File: Golang/1-50/17.go
package main

import (
	"fmt"
)

func fff(n int) int {
	s := 0
	for n > 0 {
		p := n % 10
		s = s + p
		n = n / 10
	}
	return s * s
}

func main() {
	var n int
	fmt.Scan(&n)
	k := 0
	i := 1
	var h int
	for k < n {
		if i%fff(i) == 0 {
			k++
		}
		h = i
		i++
	}
	fmt.Println(h)
}
